prototypes/javaspec-jupiter/src/test/java/info/javaspec/jupiter/syntax/fixture/JavaSpec.java

```diff
@@ -0,0 +1,161 @@
+package info.javaspec.jupiter.syntax.fixture;
+
+import org.junit.jupiter.api.DynamicContainer;
+import org.junit.jupiter.api.DynamicNode;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.function.Executable;
+
+import java.util.ArrayDeque;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+final class JavaSpec<S> {
+  private final Deque<DynamicNodeList> containers = new ArrayDeque<>();
+  private Supplier<S> subjectSupplier;
+
+  public JavaSpec() {
+    //Push a null object onto the bottom of the stack, so there's always a parent list to add nodes to.
+    //Unlike all other entries on the stack, the root node list does not get turned into a DynamicContainer.
+    containers.addLast(new RootNodeList());
+  }
+
+  //Positive: Supports any kind of code, that needs to run after each spec.
+  public void afterEach(Executable clean) {
+    containers.peekLast().setAfterEach(clean);
+  }
+
+  //Positive: Supports any kind of code, that needs to run before each spec.
+  public void beforeEach(Executable arrange) {
+    containers.peekLast().setBeforeEach(arrange);
+  }
+
+  public void context(String condition, ContextBlock block) {
+    declareContainer(condition, block);
+  }
+
+  public DynamicNode describe(Class<?> actor, DescribeBlock block) {
+    return declareContainer(actor.getSimpleName(), block);
+  }
+
+  public DynamicNode describe(String functionOrGenericActor, DescribeBlock block) {
+    return declareContainer(functionOrGenericActor, block);
+  }
+
+  public DynamicTest it(String behavior, Executable verification) {
+    //Positive: The algorithm for collecting fixture lambdas is straightforward enough, using a Deque.
+    List<Executable> arrangements = containers.stream()
+      .map(x -> x.arrange)
+      .filter(Objects::nonNull)
+      .collect(Collectors.toList());
+
+    List<Executable> cleaners = containers.stream()
+      .map(x -> x.cleanup)
+      .filter(Objects::nonNull)
+      .collect(Collectors.toList());
+    Collections.reverse(cleaners);
+
+    return containers.peekLast().addTest(behavior, arrangements, verification, cleaners);
+  }
+
+  public DynamicNode pending(String pendingBehavior) {
+    DynamicTest test = makeSkippedTest(
+      pendingBehavior,
+      String.format("Pending: %s.  This is not a failed assumption in the spec; it's just how JavaSpec skips a pending a spec.", pendingBehavior)
+    );
+
+    addToCurrentContainer(test);
+    return test;
+  }
+
+  public S subject() {
+    return this.subjectSupplier.get();
```

Comment:
I noticed an issue when I made the example code, where `beforeEach` and `it` both call `#subject()`.

- [ ] Memoize the value once it has been supplied and clear it out at the end of the spec, so that separate calls to `#subject.get()` return the same instance.